Content.Server/Drone/DroneSystem.cs

```diff
@@ -0,0 +1,145 @@
+using Content.Shared.Drone;
+using Content.Server.Drone.Components;
+using Content.Shared.Drone.Components;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Inventory.Events;
+using Content.Shared.MobState.Components;
+using Content.Shared.MobState;
+using Content.Shared.DragDrop;
+using Content.Shared.Examine;
+using Content.Server.Popups;
+using Content.Server.Mind.Components;
+using Content.Server.Ghost.Roles.Components;
+using Content.Server.Hands.Components;
+using Content.Server.Explosion.EntitySystems;
+using Content.Server.Explosion;
+using Content.Server.Actions.Events;
+using Robust.Shared.IoC;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Localization;
+using Robust.Shared.Player;
+using Content.Shared.Tag;
+
+
+namespace Content.Server.Drone
+{
+    public class DroneSystem : SharedDroneSystem
+    {
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly ExplosionSystem _explosions = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<DroneComponent, MobStateChangedEvent>(OnMobStateChanged);
+            SubscribeLocalEvent<DroneComponent, DisarmAttemptEvent>(OnDisarmAttempt);
+            SubscribeLocalEvent<DroneComponent, DropAttemptEvent>(OnDropAttempt);
+            SubscribeLocalEvent<DroneComponent, IsUnequippingAttemptEvent>(OnUnequipAttempt);
+            SubscribeLocalEvent<DroneComponent, ExaminedEvent>(OnExamined);
+            SubscribeLocalEvent<DroneComponent, MindAddedMessage>(OnMindAdded);
+            SubscribeLocalEvent<DroneComponent, MindRemovedMessage>(OnMindRemoved);
+        }
+
+        private void OnExamined(EntityUid uid, DroneComponent component, ExaminedEvent args)
+        {
+            if (args.IsInDetailsRange)
+            {
+                if (EntityManager.TryGetComponent<MindComponent>(uid, out var mind) && mind.HasMind)
+                {
+                    args.PushMarkup(Loc.GetString("drone-active"));
+                }
+                else
+                {
+                    args.PushMarkup(Loc.GetString("drone-dormant"));
+                }
+            }
+        }
+
+        private void OnMobStateChanged(EntityUid uid, DroneComponent drone, MobStateChangedEvent args)
+        {
+            if (args.Component.IsDead())
+            {
```

Comment:
cant you just check the new state it gives you directly in the args

Reply:
I mean IMobState also just uses IsDead() and I'd have to add a bunch of nullability shit